backend/src/main/java/nl/itvitae/twitbook/user/UserController.java

```diff
@@ -20,19 +24,46 @@
 public class UserController {
 
   private final UserRepository userRepository;
+  private final FollowRepository followRepository;
 
   @GetMapping
   public List<UserDTO> getAll() {
     return userRepository.findAll().stream().map(UserDTO::new).toList();
   }
 
   @GetMapping("by-username/{username}")
-  public ResponseEntity<?> findByUsername(@PathVariable String username) {
+  public ResponseEntity<?> findByUsername(@PathVariable String username,  @AuthenticationPrincipal User authUser) {
     Optional<User> user = userRepository.findByUsernameIgnoreCase(username);
 
-    if(user.isEmpty())
+    if (user.isEmpty()) {
       return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+    }
 
-    return new ResponseEntity<>(new UserDTO(user.get()), HttpStatus.OK);
+    if (authUser == null) {
+      return new ResponseEntity<>(new UserDTO(user.get()), HttpStatus.OK);
+    }
+    return new ResponseEntity<>(new UserDTO(user.get(), followRepository.existsFollowByFollowerIdAndFollowingId(authUser.getId(), user.get().getId())), HttpStatus.OK);
+  }
+
+  @PostMapping("/follow/{followingUsername}")
```

Comment:
Move this to the FollowController, e.g. `/follow/user/{username}` and add that to the securityconfig